[import/no-duplicates] (prefer-inline) Autofix merges imports incorrectly

<ins>**Config:**</ins>
```typescript
'import/no-duplicates': ['error', { 'prefer-inline': true }]
```

<ins>**Case 1:**</ins>

input
```typescript
import type { TypeOne, TypeTwo } from './types';
import type { TypeThree, TypeFour } from './types';
```

output
```typescript
import type { TypeOne, TypeTwo, type TypeThree, TypeFour } from './types';
```

expected
```typescript
import { type TypeOne, type TypeTwo, type TypeThree, type TypeFour } from './types';
```

<ins>**Case 2:**</ins>

input
```typescript
import type { TypeOne, TypeTwo } from './types';
import { type TypeThree, type TypeFour } from './types';
```

output
```typescript
import type { TypeOne, TypeTwo, type TypeThree, type TypeFour } from './types';
```

expected
```typescript
import { type TypeOne, type TypeTwo, type TypeThree, type TypeFour } from './types';
```

Ran into this today. Original code:

```typescript
import type { Knex } from 'knex';
import knex from 'knex';
```

Produces these warnings:

```
index.ts:3:27
✖    3:27  /project/node_modules/knex/knex.js imported multiple times.    import/no-duplicates
✖    4:18  /project/node_modules/knex/knex.js imported multiple times.    import/no-duplicates
```

Auto-fixes to:

```typescript
import knex, type { Knex } from 'knex';
```

Then when running eslint again:

```
index.ts:3:13
✖  3:13  Parsing error: { expected.  

1 error
```

And also produces the following TypeScript error:

```
index.ts:3:28 - error TS1434: Unexpected keyword or identifier.

3 import knex, type { Knex } from 'knex';
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ This change log adheres to standards from [Keep a CHANGELOG](https://keepachange
 ### Fixed
 - [`no-extraneous-dependencies`]: allow wrong path ([#3012], thanks [@chabb])
 - [`no-cycle`]: use scc algorithm to optimize ([#2998], thanks [@soryy708])
+- [`no-duplicates`]: Removing duplicates breaks in TypeScript ([#3033], thanks [@yesl-kim])
 
 ### Changed
 - [Docs] `no-extraneous-dependencies`: Make glob pattern description more explicit ([#2944], thanks [@mulztob])
@@ -1121,6 +1122,7 @@ for info on changes for earlier releases.
 
 [`memo-parser`]: ./memo-parser/README.md
 
+[#3033]: https://github.com/import-js/eslint-plugin-import/pull/3033
 [#3012]: https://github.com/import-js/eslint-plugin-import/pull/3012
 [#3011]: https://github.com/import-js/eslint-plugin-import/pull/3011
 [#3004]: https://github.com/import-js/eslint-plugin-import/pull/3004
@@ -1962,6 +1964,7 @@ for info on changes for earlier releases.
 [@wtgtybhertgeghgtwtg]: https://github.com/wtgtybhertgeghgtwtg
 [@xM8WVqaG]: https://github.com/xM8WVqaG
 [@xpl]: https://github.com/xpl
+[@yesl-kim]: https://github.com/yesl-kim
 [@yndajas]: https://github.com/yndajas
 [@yordis]: https://github.com/yordis
 [@Zamiell]: https://github.com/Zamiell

--- a/src/rules/no-duplicates.js
+++ b/src/rules/no-duplicates.js
@@ -132,6 +132,7 @@ function getFix(first, rest, sourceCode, context) {
   const shouldAddDefault = getDefaultImportName(first) == null && defaultImportNames.size === 1;
   const shouldAddSpecifiers = specifiers.length > 0;
   const shouldRemoveUnnecessary = unnecessaryImports.length > 0;
+  const preferInline = context.options[0] && context.options[0]['prefer-inline'];
 
   if (!(shouldAddDefault || shouldAddSpecifiers || shouldRemoveUnnecessary)) {
     return undefined;
@@ -157,8 +158,7 @@ function getFix(first, rest, sourceCode, context) {
       ([result, needsComma, existingIdentifiers], specifier) => {
         const isTypeSpecifier = specifier.importNode.importKind === 'type';
 
-        const preferInline = context.options[0] && context.options[0]['prefer-inline'];
-        // a user might set prefer-inline but not have a supporting TypeScript version.  Flow does not support inline types so this should fail in that case as well.
+        // a user might set prefer-inline but not have a supporting TypeScript version. Flow does not support inline types so this should fail in that case as well.
         if (preferInline && (!typescriptPkg || !semver.satisfies(typescriptPkg.version, '>= 4.5'))) {
           throw new Error('Your version of TypeScript does not support inline type imports.');
         }
@@ -185,6 +185,18 @@ function getFix(first, rest, sourceCode, context) {
     );
 
     const fixes = [];
+
+    if (shouldAddSpecifiers && preferInline && first.importKind === 'type') {
+      // `import type {a} from './foo'` → `import {type a} from './foo'`
+      const typeIdentifierToken = tokens.find((token) => token.type === 'Identifier' && token.value === 'type');
+      fixes.push(fixer.removeRange([typeIdentifierToken.range[0], typeIdentifierToken.range[1] + 1]));
+
+      tokens
+        .filter((token) => firstExistingIdentifiers.has(token.value))
+        .forEach((identifier) => {
+          fixes.push(fixer.replaceTextRange([identifier.range[0], identifier.range[1]], `type ${identifier.value}`));
+        });
+    }
 
     if (shouldAddDefault && openBrace == null && shouldAddSpecifiers) {
       // `import './foo'` → `import def, {...} from './foo'`

--- a/tests/src/rules/no-duplicates.js
+++ b/tests/src/rules/no-duplicates.js
@@ -705,6 +705,24 @@ context('TypeScript', function () {
           ],
         }),
         test({
+          code: "import type {x} from 'foo'; import {type y} from 'foo'",
+          ...parserConfig,
+          options: [{ 'prefer-inline': true }],
+          output: `import {type x,type y} from 'foo'; `,
+          errors: [
+            {
+              line: 1,
+              column: 22,
+              message: "'foo' imported multiple times.",
+            },
+            {
+              line: 1,
+              column: 50,
+              message: "'foo' imported multiple times.",
+            },
+          ],
+        }),
+        test({
           code: "import {type x} from 'foo'; import type {y} from 'foo'",
           ...parserConfig,
           output: `import {type x,y} from 'foo'; `,

